libp2p/services/hpservice.nim

```diff
@@ -63,6 +63,31 @@ proc tryStartingDirectConn(self: HPService, switch: Switch, peerId: PeerId): Fut
       continue
   return false
 
+proc newConnectedPeerHandler(self: HPService, switch: Switch, peerId: PeerId, event: PeerEvent) {.async.} =
+  try:
+    # Get all connections to the peer. If there is at least one non-relayed connection, return.
+    var relayedConnt: Connection
+    for muxer in switch.connManager.getConnections()[peerId]:
+      let conn = muxer.connection
+      if not isRelayed(conn):
+        return
+      elif conn.transportDir == Direction.In:
+        relayedConnt = conn
+
+    if not isNil(relayedConnt):
```

Comment:
```suggestion
    let connections = switch.connManager.getConnections()[peerId].mapIt(it.connection)
    if connections.anyIt(not isRelayed(it)):
      return
    let incomingRelays = connections.filterIt(it.transportDir == Direction.In)
    if incomingRelays.len == 0:
      return
    
    let relayedConn = incomingRelays[0]
```
(need to de-ident the rest of the code)
wdyt?

Reply:
Not sure if this is better

Reply:
I think the current version is slightly confusing, I tried to convey the intention more clearly

Reply:
done